what4/src/What4/Expr/Allocator.hs

```diff
@@ -0,0 +1,197 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes #-}
+
+{-|
+Module      : What4.Expr.Allocator
+Description : Expression allocators for controlling hash-consing
+Copyright   : (c) Galois Inc, 2015-2022
+License     : BSD3
+Maintainer  : rdockins@galois.com
+-}
+module What4.Expr.Allocator
+( ExprAllocator(..)
+, newStorage
+, newCachedStorage
+
+, cacheStartSizeOption
+, cacheStartSizeDesc
+, cacheTerms
+, cacheOptDesc
+) where
+
+import           Control.Lens ( (&) )
+import           Control.Monad.ST (stToIO)
+import           Data.IORef
+
+import qualified Data.Parameterized.HashTable as PH
+import           Data.Parameterized.Nonce
+
+import           What4.BaseTypes
+import           What4.Concrete
+import           What4.Config as CFG
+import           What4.Expr.App
+import           What4.ProgramLoc
+import           What4.Utils.AbstractDomains
+
+------------------------------------------------------------------------
+-- Cache start size
+
+-- | Starting size for element cache when caching is enabled.
+--
```

Comment:
Size in .. ?  bytes?  elements?

I realize these were just moved, but seems like a good opportunity for clarification.